app/models/exam.rb

```diff
@@ -141,8 +181,13 @@ def self.upsert_students!(json)
   def self.adapt_json_values(exam)
     exam[:guide_id] = Guide.locate!(exam[:slug]).id
     exam[:organization_id] = Organization.current.id
+    exam[:course_id] = Course.locate!(exam[:course].to_s).id
     exam[:users] = User.where(uid: exam[:uids])
-    [:start_time, :end_time].each { |param| exam[param] = exam[param].to_time }
+    exam[:start_time] = exam[:start_time].to_time
+    exam[:end_time] = exam[:end_time].to_time
+    exam[:classroom_id] = exam[:eid] if exam[:eid].present?
+    exam[:passing_criterion_type] = exam.dig(:passing_criterion, :type)
+    exam[:passing_criterion_value] = exam.dig(:passing_criterion, :value)
```

Comment:
Perhaps we could use plain setters here: 

```ruby
exam.guide = Guide.locate!(exam.slug)
exam.organization = Organization.current
exam.course = Course.locate!(exam.course.to_s)
# and so on...
```

Reply:
Yep, I agree. I would leave it for the next PR though, as it actually unifies everything and makes this change more straight forward